From 11ebc884041874ad8ea799ee224d740254550baf Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Mon, 28 Jul 2014 14:46:31 -0700
Subject: [PATCH 085/117] mt9m114: Fix resolution index initiailization

Initializing to an "invalid" token is wrong.  V4L2 provides no way to
expose the idea that the sensor is in an uninitialized state and will
accept ioctls on it assuming whatever defaults are present.  Worse,
the driver wouldn't check this initialization cookie and would panic
if the call to _g_bin_factor_x() arrived before the call to
_set_mbus_format().  Even worse: when using GMS camera apps, it does.

Leave the value at the zero it gets from kzalloc, thus leaving the
default resolution at the first entry in the table.

Change-Id: Id4c7f8eb66b1636c73f89f8c8211f6262f6820c0
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c b/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
index e986083..12ba009 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c
@@ -1875,9 +1875,6 @@ static int mt9m114_probe(struct i2c_client *client,
 		return ret;
 	}
 
-	/* set res index to be invalid */
-	dev->res = -1;
-
 	return 0;
 }
 
-- 
1.9.3

